agent/engine/task_manager.go

```diff
@@ -732,8 +799,42 @@ func (mtask *managedTask) startContainerTransitions(transitionFunc containerTran
 	return anyCanTransition, transitions, reasons
 }
 
+// startResourceTransitions steps through each resource in the task and calls
+// the passed transition function when a transition should occur
+func (mtask *managedTask) startResourceTransitions(transitionFunc resourceTransitionFunc) (bool, map[string]taskresource.ResourceStatus) {
+	anyCanTransition := false
+	transitions := make(map[string]taskresource.ResourceStatus)
+	for _, res := range mtask.Resources {
+		transition := mtask.resourceNextState(res)
+		if transition.reason != nil {
```

Comment:
The `reason` here indicates the error why the resource cannot be transitioned. 
Not all resource transitions require 'action'. This is indicated by the `actionRequired` bool. Hence the two different variables.

Reply:
But, it's not being used anywhere. Right now, you can use them interchangeably. I'm not sure what value `reason` is providing right now.

Reply:
These are the possible values of the variables and how they are used now:
`reason`:
`nil`- resource's known status can change(this can happen by applying a transition function and changing the known status OR just changing the known status without applying any transition).
not `nil` - resource's known status need not change and the reason why. 

`actionRequired`:
`true` - known status can be changed (given reason is `nil`) by applying a transition function. 
`false` - known status can be changed (given reason is `nil`) without applying a transition function.